lib/repo/git_repository.rb

```diff
@@ -524,36 +645,45 @@ def self.add_user(user_id, permissions, repo_name)
     def self.set_bulk_permissions(repo_names, user_id_permissions_map)
       # Check if configuration is in order
       if Repository.conf[:IS_REPOSITORY_ADMIN].nil?
-        raise ConfigurationError.new("Required config 'IS_REPOSITORY_ADMIN' not set")
+        raise ConfigurationError.new(
+          "Required config 'IS_REPOSITORY_ADMIN' not set")
       end
       # If we're not in authoritative mode, bail out
       if !Repository.conf[:IS_REPOSITORY_ADMIN] # Are we admin?
-        raise NotAuthorityError.new("Unable to set bulk permissions:  Not in authoritative mode!");
+        raise NotAuthorityError.new(
+          'Unable to set bulk permissions: Not in authoritative mode!')
       end
 
-      # Check if gitolite admin repo exists, create it if not
-      if !Gitolite::GitoliteAdmin.is_gitolite_admin_repo?(Repository.conf[:REPOSITORY_STORAGE])
-        Gitolite::GitoliteAdmin.bootstrap(Repository.conf[:REPOSITORY_STORAGE])
-      end
+      ga_repo = Gitolite::GitoliteAdmin.new(
+        Repository.conf[:REPOSITORY_STORAGE] +
+          '/gitolite-admin', GITOLITE_SETTINGS)
+
+      # Sync admin repo
+      ga_repo.update
 
-      #gitolite admin repo
-      ga_repo = Gitolite::GitoliteAdmin.new(Repository.conf[:REPOSITORY_STORAGE])
+      # The admin repo is loaded into memory
       conf = ga_repo.config
 
       repo_names.each do |repo_name|
         repo_name = File.basename(repo_name)
         repo = ga_repo.config.get_repo(repo_name)
         if repo.nil?
+          # Generate new repo conf since this repo hasn't been created yet
           repo = Gitolite::Config::Repo.new(repo_name)
         end
+        # Add the permissions for each user
         user_id_permissions_map.each do |user_id, permissions|
           perm_string = __translate_to_git_perms(permissions)
           repo.add_permission(perm_string, "", user_id)
+          repo.add_permission(perm_string, "", "git") # testing
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.